lua/neo-tree/sources/common/commands.lua

```diff
@@ -11,34 +11,42 @@ local log = require("neo-tree.log")
 local help = require("neo-tree.sources.common.help")
 local Preview = require("neo-tree.sources.common.preview")
 
----Gets the node parent folder recursively
----@param tree table to look for nodes
----@param node table to look for folder parent
+---Gets the node parent folder
+---@param state table to look for nodes
 ---@return table table
-local function get_folder_node(state, node)
+local function get_folder_node(state)
   local tree = state.tree
-  if not node then
-    node = tree:get_node()
-  end
+  local node = tree:get_node()
+  local last_id = node:get_id()
+
+  -- arbitrary number to avoid infinite loop
+  while true do
```

Comment:
I think it would be more correct to be:
```suggestion
  while node do
```
because the code below would fail with an error if `node` was nil.

Reply:
wouldn't that just make the function return `nil` and error somewhere else instead? making the change throws off my lsp and it asks me to add a return value

Reply:
also I don't think anything in the loop would make `node` `nil` unless `get_parent_id` returns an invalid id

Reply:
In _theory_ [this line](https://github.com/nvim-neo-tree/neo-tree.nvim/blob/f5ed1fd361c820b6d30fc77cc3a6d33027eff1bf/lua/neo-tree/sources/common/commands.lua#L46) could result in node being nil if `parent_id` is nonsensical (for some reason). This is because [get_node](https://github.com/MunifTanjim/nui.nvim/tree/main/lua/nui/tree#treeget_node) returns `nil` if it cant find a matching node for the id provided.

Will this happen? No probably not. But setting the loop to rely on `node` being _something_ provides **another** safe guard to avoid dry looping (and to a degree makes the code just a bit more clear on what it is doing).

It doesn't hurt to set the loop to rely on node instead of true, and does arguably provide benefit, no reason not to do it

Reply:
changed it to `while node` and updated the type signature to make my editor happy